x/wasm/internal/types/msg.go

```diff
@@ -182,3 +191,39 @@ func (msg MsgExecuteContract) GetSignBytes() []byte {
 func (msg MsgExecuteContract) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{msg.Sender}
 }
+
+type MsgMigrateContract struct {
+	Sender   sdk.AccAddress  `json:"sender" yaml:"sender"`
+	Contract sdk.AccAddress  `json:"contract" yaml:"contract"`
+	Code     uint64          `json:"code_id" yaml:"code_id"`
+	Msg      json.RawMessage `json:"msg" yaml:"msg"` // TODO: Rename to parameters? this is passed to the contract's migration method and can contain custom parameters
```

Comment:
In cosmwasm contracts, we used the word Params/Env to represent the blockchain state (kind of like sdk.Context)
And use the term Msg is data sent to and from contracts.

That  said, you could rename this `MigrateMsg` to be more explicit.